### PR TITLE
Add a job for clang in Ubuntu 18.04

### PR DESCRIPTION
**Pull request by shrijitsingh99**
As discussed in [#4081](https://github.com/PointCloudLibrary/pcl/pull/4081#issuecomment-626897659)

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Shall we reorganize the CI based on compilers?
* GCC: Ubuntu 16, 20
* Clang: MacOS 10.14, 10.15, Ubuntu 18
* MSVC: Windows

This will allow Ubuntu 18 CI to run after (or before) MacOS, reducing total parallel needed, as well as not increasing CI time (MacOS CI is faster than tutorial CI)

**Comment by shrijitsingh99** [comment]
> This will allow Ubuntu 18 CI to run after (or before) MacOS, reducing total parallel needed, as well as not increasing CI time (MacOS CI is faster than tutorial CI)

Oh yeah, I forgot about that part of the discussion 😅 
Sounds good putting it after macOS.

File: .ci/azure-pipelines/azure-pipelines.yaml [hunk 1]
```diff
@@ -14,29 +16,113 @@ stages:
     displayName: Formatting
     jobs:
       - template: formatting.yaml
-  - stage: build_ubuntu
-    displayName: Build Ubuntu
+
+  - stage: build_gcc
+    displayName: Build GCC
```
Comment (kunaltyagi) [inline .ci/azure-pipelines/azure-pipelines.yaml:21]:
```suggestion
    displayName: GCC
```

**Comment by kunaltyagi** [comment]
> const variables are implicitly shared

Errrr. not quiet. See @SunBlack's PR regarding OMP since this is a messed up situation

**Comment by shrijitsingh99** [comment]
> > const variables are implicitly shared
> 
> Errrr. not quite. See @SunBlack's PR regarding OMP since this is a messed up situation

Thanks, that makes it clearer. I was going off [this](http://jakascorner.com/blog/2016/07/omp-default-none-and-const.html) initially, guess that's outdated. The GCC 9 porting [guide](https://gcc.gnu.org/gcc-9/porting_to.html) also has this (for future reference).

What still is unclear why this issue even came up since initially the variables `s` and `max_dist_sqr` were explicitly declared as `firstprivate` which should not have caused an error right?

File: .ci/azure-pipelines/azure-pipelines.yaml (hunk 1 above)
Comment (shrijitsingh99) [reply .ci/azure-pipelines/azure-pipelines.yaml:21]: Do this for the rest too?

Comment (shrijitsingh99) [reply .ci/azure-pipelines/azure-pipelines.yaml:21]: I think we should keep the name `Build GCC`  since it implies it is the build stage in the pipeline.

Comment (kunaltyagi) [reply .ci/azure-pipelines/azure-pipelines.yaml:21]: If you look at the synopsis shown, it seems redundant.  But if you look at the [extended information](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=16460&view=results), it seems necessary.

Screen Shot 2020-05-25 at 8 06 06

File: .ci/azure-pipelines/azure-pipelines.yaml [hunk 2]
```diff
@@ -14,29 +16,114 @@ stages:
...
+          CMAKE_CXX_FLAGS: '-Wall -Wextra -Werror -Wabi'
+        steps:
+          - template: build/build-macos.yaml
+      - job: ubuntu
+        displayName: Ubuntu
+        dependsOn: osx
+        condition: succeededOrFailed()
+        pool:
+          vmImage: 'Ubuntu 16.04'
+        strategy:
```
Comment (kunaltyagi) [inline .ci/azure-pipelines/azure-pipelines.yaml:77]: One or two lines explaining the reasoning behind the way the tests are run would be helpful. Eg: allows Mac, Windows and Ubuntu to be run in parallel. Placement of Ubuntu-clang job reduces need of parallel CI jobs and doesn't impact total time per run under ideal conditions.

**Comment by SergioRAgostinho** [comment]
There's a `build/` folder prefixing the scripts. It might be worth to strip out the `build-` prefix in the names of the files.

File: .ci/azure-pipelines/azure-pipelines.yaml (hunk 1 above)
Comment (shrijitsingh99) [reply .ci/azure-pipelines/azure-pipelines.yaml:21]: The Github one can be changed I think if we properly integrate with Github workflow.
[Here](https://github.com/github/VisualStudio/pull/2512/checks?check_run_id=653837445) they have seemed to integrate properly with workflow. 
Changing it also might create problems if we have a separate stage for text in the future 

It's not a big issue anyway will leave finally call to you.

Comment (kunaltyagi) [reply .ci/azure-pipelines/azure-pipelines.yaml:21]: > It's not a big issue anyway

Yeah. Plus that example is of GitHub actions. Something very different (and quite similar)

**Pull request merged**